src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Duke {
+
+    private static final String filePath = "data/duke.txt";
+    private static final String fileFolder = "data";
+
+    public static void main(String[] args) throws UnknownInputException, IOException {
+        //create folder if needed
+        File folder = new File(fileFolder);
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+
+        //create file if needed
+        File f = new File(filePath);
+        if (!f.exists()) {
+            f.createNewFile();
+        }
+
+        //create tasks arraylist from file
+        //tasks is an array list collection of task objects
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File dataFile = new File(filePath);
+            Scanner scanner = new Scanner(dataFile);
+
+            while (scanner.hasNext()) {
+                String line = scanner.nextLine();
+                String category = line.substring(0, 1);
+                String descriptor = line.substring(8);
+                int dateIndex = descriptor.indexOf('|');
+                switch (category) {
+                    case "T":
+                        tasks.add(new Todo(descriptor));
+                        break;
+                    case "E":
+                        tasks.add(new Event(descriptor.substring(0, dateIndex - 1), descriptor.substring(dateIndex + 2)));
```

Comment:
Should split into multiple lines. Also consider doing the parsing such that you don't need explicit indices (e.g. using a trim whitespace method).